File: rust/search/src/linear.rs
pub fn linear_search<T: PartialEq>(list: &Vec<T>, needle: T) -> bool {
    for item in list {
        if *item == needle {
            return true;
        }
    }

    false
}

#[cfg(test)]
mod tests {

    use super::linear_search;

    #[test]
    fn linear_search_array() {
        let foo = vec![1, 3, 4, 69, 71, 81, 90, 99, 420, 1337, 69420];

        assert!(linear_search(&foo, 69));
        assert!(!linear_search(&foo, 1336));
        assert!(linear_search(&foo, 69420));
        assert!(!linear_search(&foo, 69421));
        assert!(linear_search(&foo, 1));
        assert!(!linear_search(&foo, 0));
    }
}

File: zig/src/search/linear.zig
const std = @import("std");
const expect = std.testing.expect;

fn linear_search(haystack: []u32, needle: u32) bool {
    for (haystack) |el| {
        if (el == needle) {
            return true;
        }
    }

    return false;
}

test "Linear search" {
    var list = [_]u32{ 9, 3, 7, 4, 69, 420, 42 };

    try expect(linear_search(&list, 42) == true);
    try expect(linear_search(&list, 1) == false);
}
